### ISSUE TITLE
[1.12-rc2] docker should provide a way to track activity

### ISSUE THREAD

**Issue by thaJeztah** [docker/docker#23706]
It's currently difficult to find out why tasks are not scheduled, other than looking at the daemon logs of the manager node.

For example;

```
docker service create --name web -p 8080:80 nginx:alpine
docker service create --name web2 -p 8080:80 nginx:alpine

docker service ls

ID            NAME  REPLICAS  IMAGE         COMMAND
d53x7mh2x6xt  web   1/1       nginx:alpine
eri5bnwygggh  web2  0/1       nginx:alpine

docker service tasks web2
ID                         NAME    SERVICE  IMAGE         LAST STATE     DESIRED STATE  NODE
ao8bdynjskrgxbpo20s7sgu60  web2.1  web2     nginx:alpine  New 2 minutes  Running        moby
```

Service `web2` has conflicting ports with service `web`, so tasks cannot be scheduled. However, as a user, I cannot see why.

I think we should have a way to see the scheduler activity

Adding this to the milestone for discussion

**Comment by dgageot** [comment]
Likewise, creating a service with a non-existing image name leaves the user without feedback on why no task is ever started. The task alternates `Accepted` and `Preparing` states.
This is specially annoying when the non-existing image name is due to a typo :-)

**Comment by stevvooe** [comment]
This is related to #23811. Please see my comment https://github.com/docker/docker/issues/23811#issuecomment-227869088.

**Comment by thaJeztah** [comment]
Is someone working on the error-reporting / including errors in `inspect` output?

**Comment by RangerMauve** [comment]
It was very frustrating to try and figure out what was wrong since the only feedback I had from my service not running is that it had `0/1` replicas up. Luckily I thought to look in the docker daemon's log and found an error, but it was really painful to get to that point.

**Comment by thaJeztah** [comment]
@RangerMauve thanks for your feedback; we agree and will be working on a solution for 1.12

**Comment by tiborvass** [comment]
This has been closed by #24917
